src/main.rs

```diff
@@ -1376,13 +1377,126 @@ fn parse_manpage_at_path(
 #[derive(Copy, Clone, Debug)]
 struct Progress(pub bool);
 
-// TODO Arg/output types?
-fn parse_and_output_man_pages(
-    _paths: impl Iterator<Item = PathBuf>,
-    _output_directory: PathBuf,
-    Progress(_show_progress): Progress,
-) {
-    unimplemented!();
+// Output: Result<cmd_name, error>
+// TODO: Result<cmd_name, CompletionsError>
+fn parse_and_output_man_pages<P: AsRef<Path>>(
+    paths: impl Iterator<Item = P>,
+    output_directory: P,
+    Progress(show_progress): Progress,
```

Comment:
I don't think we need `Progress`, we can just `bool`.

Reply:
[This](https://github.com/rust-dc/fish-manpage-completions/commit/a5270bcd0e9a98b3da2d55aabff1689e67a443d1) was a @scooter-dangle thing

Reply:
Especially when there are multiple arguments to a function of the same type and where order isn't obvious, it's a pattern for trivially ensuring the caller doesn't accidentally mix up the argument order. Not sure though why I initially added it when there was only one boolean arg to that function. :man_shrugging:

Reply:
The easiest way is to just borrow the whole options from clap arguments.

Reply:
It's removed. We can add it back if and when we add them for the rest of the bools